### ISSUE TITLE
[6.1.1] Bug: state.registeredFields not keeping track of number of calls

### ISSUE THREAD

**Issue by huan086** [erikras/redux-form#2016]
Suppose you have 2 Field referring to the same property

``` jsx
let showSecondTime = true;
<Field name="theProp" component="input" type="text" />
{showSecondTime && <Field name="theProp" component="input" type="text" />}
```

when showSecondTime is set to false subsequently, the 2nd Field get unmounted, and that clears registeredFields.theProp. However, we still need that for the first Field!
# Proposed solution

keep track of the registration count

**Comment by huan086** [comment]
Related: https://github.com/erikras/redux-form/issues/1990

**Comment by erikras** [comment]
Published in [`v6.2.0`](https://github.com/erikras/redux-form/releases/tag/v6.2.0).

**Comment by huan086** [comment]
Still wrong. Hopefully i'll have some time to fix this

**Comment by sergi** [comment]
@huan086 any luck with this? I am still experiencing with v6.2.0

**Comment by huan086** [comment]
@sergi No... Don't think I'll have time to fix it myself until next year. Could you to give it a shot at fixing?

**Comment by sergi** [comment]
@huan086 Unfortunately not until next year...

**Comment by halt-hammerzeit** [comment]
@sergi @huan086 Btw I wrote my alternative to `redux-form` which fixes this issue.
Or maybe someone coming here from google would be interested.
https://github.com/halt-hammerzeit/simpler-redux-form

**Comment by huan086** [comment]
@halt-hammerzeit I've seen your library. The library don't have the features that I need. Would be great if you put your effort in improving redux-form instead. As for the bloat, don't worry. redux-form is written in ES6, so a bundler like rollup will totally remove the features you didn't use

**Comment by halt-hammerzeit** [comment]
@huan086 And what are the features you wanted that aren't present in my library?

**Comment by huan086** [comment]
@halt-hammerzeit Fields, FieldArray, form level validation, submit success/fail callbacks, place for me to hook custom reducer onto form reducer (I wrote a custom plugin that does splice that accepts multiple inserts to overcome the limitation of redux-form splice), possibly many other callbacks that I can't remember that redux-form provides to allow me to modify its behavior

Basically, it's a large form that redux-form v5 is also not suitable...

**Comment by erikras** [comment]
Fix released as [`v6.5.0`](https://github.com/erikras/redux-form/releases/tag/v6.5.0).
